test/helpers.rb

```diff
@@ -4,17 +4,17 @@
 require "tempfile"
 
 module Helpers
-  TEST_OBJ = "test/bin/hello.o"
-  TEST_EXE = "test/bin/hello.bin"
-  TEST_DYLIB = "test/bin/libhello.dylib"
-  TEST_EXTRA_DYLIB = "test/bin/libextrahello.dylib"
-  TEST_BUNDLE = "test/bin/hellobundle.so"
-
-  TEST_FAT_OBJ = "test/bin/fathello.o"
-  TEST_FAT_EXE = "test/bin/fathello.bin"
-  TEST_FAT_DYLIB = "test/bin/libfathello.dylib"
-  TEST_FAT_EXTRA_DYLIB = "test/bin/libfatextrahello.dylib"
-  TEST_FAT_BUNDLE = "test/bin/fathellobundle.so"
+  TEST_OBJ = "test/bin/x86/hello.o"
+  TEST_EXE = "test/bin/x86/hello.bin"
+  TEST_DYLIB = "test/bin/x86/libhello.dylib"
+  TEST_EXTRA_DYLIB = "test/bin/x86/libextrahello.dylib"
+  TEST_BUNDLE = "test/bin/x86/hellobundle.so"
+
+  TEST_FAT_OBJ = "test/bin/x86/fathello.o"
+  TEST_FAT_EXE = "test/bin/x86/fathello.bin"
+  TEST_FAT_DYLIB = "test/bin/x86/libfathello.dylib"
+  TEST_FAT_EXTRA_DYLIB = "test/bin/x86/libfatextrahello.dylib"
+  TEST_FAT_BUNDLE = "test/bin/x86/fathellobundle.so"
```

Comment:
Maybe rename the constants and use `i386` or `x86_64` instead of of `x86` for consistency with `man 3 arch` as pointed out in https://github.com/Homebrew/ruby-macho/pull/24#issuecomment-223326092. For fat binaries I'd even suggest to put them in a separate directory, e.g. `fat-i386-ppc` would be used for a classic universal binary. `fat-i386-x86_64` would be used for a more modern 32/64-bit Intel universal binary.

Another idea: Drop all the constants and instead use a helper method that constructs the path for you, e.g.:

```ruby
def binary_fixture(archs, name)
  arch_dir = archs.is_a?(Array) ? "fat-#{archs.join("-")}" : archs.to_s
  "test/bin/#{arch_dir}/#{name}"
end
```

It would be used something like this:

```ruby
binary_fixture(:i386, "hello.o") # Yields `test/bin/i386/hello.o`.
binary_fixture([:i386, :ppc], "libhello.dylib") # Yields `test/bin/fat-i386-ppc/libhello.dylib`.
```